File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char>set;
        int l=0,r=0,m=0;

        while(r<s.length()){
            if(set.count(s[r]) == 0){
                set.insert(s[r]);
                m = max(m,r-l+1);
            }
            else{
                while(set.count(s[r])){
                    set.erase(s[l]);
                    l++;
                }
                set.insert(s[r]);
            }
            r++;
        }
        return m;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        vis = set()
        l = 0
        c=0
        for i in range(len(s)):
            while s[i] in vis:
                vis.remove(s[l])
                l += 1
            vis.add(s[i])
            c = max(c, i-l+1)
        return c